app/src/test/java/com/x8bit/bitwarden/ui/autofill/fido2/manager/Fido2CompletionManagerTest.kt

```diff
@@ -0,0 +1,130 @@
+package com.x8bit.bitwarden.ui.autofill.fido2.manager
+
+import android.app.Activity
+import android.content.Intent
+import androidx.credentials.provider.PendingIntentHandler
+import com.x8bit.bitwarden.data.autofill.fido2.model.Fido2CredentialAssertionResult
+import com.x8bit.bitwarden.data.autofill.fido2.model.Fido2RegisterCredentialResult
+import io.mockk.MockKVerificationScope
+import io.mockk.Ordering
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkConstructor
+import io.mockk.mockkObject
+import io.mockk.runs
+import io.mockk.unmockkConstructor
+import io.mockk.unmockkObject
+import io.mockk.verify
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class Fido2CompletionManagerTest {
+
+    private val mockActivity = mockk<Activity> {
+        every { setResult(Activity.RESULT_OK, any()) } just runs
+        every { finish() } just runs
+    }
+    private val fido2CompletionManager = Fido2CompletionManagerImpl(mockActivity)
```

Comment:
Should we make an inner class for these tests and then add some coverage for the no-op implementation in a second inner class?

Reply:
```kotlin
private lateinit var fido2CompletionManager: Fido2CompletionManager

@Nested
inner class NoOp {
    @BeforeEach
    fun setup() {
        fido2CompletionManager = Fido2CompletionManagerUnsupportedApiImpl
    }
    
    // tests!
}

@Nested
inner class Implementation {
    @BeforeEach
    fun setup() {
        fido2CompletionManager = Fido2CompletionManagerImpl(mockActivity)
    }
    
    // tests!
}
```